pkg/clusteragent/clusterchecks/dispatcher_rebalance.go

```diff
@@ -163,20 +174,26 @@ func (d *dispatcher) moveCheck(src, dest, checkID string) error {
 
 // rebalance tries to optimize the checks repartition on cluster level check
 // runners with less possible check moves based on the runner stats.
-func (d *dispatcher) rebalance() {
+func (d *dispatcher) rebalance() []RebalancingDecision {
 	start := time.Now()
 	defer func() {
 		rebalancingDuration.Set(time.Since(start).Seconds(), le.JoinLeaderValue)
 	}()
 
+	// Collect CLC runners stats and update cache before rebalancing
+	d.updateRunnersStats()
```

Comment:
The metric `rebalancingDuration`'s purpose is to monitor the rebalancing algorithm itself regardless of the stats update duration.

We already have a metric to mesure the stats update duration in `updateRunnersStats` that can reflect external issues (network latency etc.) so I believe it's best if we don't include the stats update duration in the rebalancing duration as we want `rebalancingDuration` to reflect the perf of the rebalancing algorithm itself. In case we're interested in monitoring the whole duration we can sum the two metrics in the dashboard.

Maybe we can just call `d.updateRunnersStats()` before `start := time.Now()`, wdyt ?